hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -208,22 +145,15 @@ public WorldStateAccount get(Address address) {
 			return null;
 		}
 
-		final long expiry = ledger.expiry(id);
-		final EntityId proxy = ledger.proxy(id);
-		final long balance = ledger.getBalance(id);
-		final long autoRenewPeriod = ledger.autoRenewPeriod(id);
+		AccountState accountState = repositoryRoot.getAccountState(address.toArray());
+		final long expiry = accountState.getExpirationTime();
+		final EntityId proxy = new EntityId(accountState.getProxyAccountShard(),
+				accountState.getProxyAccountRealm(),accountState.getProxyAccountNum());
+		final long balance = accountState.getBalance().longValue();
+		final long autoRenewPeriod = accountState.getAutoRenewPeriod();
 		return new WorldStateAccount(address, Wei.of(balance), expiry, autoRenewPeriod, proxy);
 	}
 
-	public void addPropertiesFor(Address address, String memo, JKey key, Id proxyAccount) {
-		WorldStateAccount account = this.provisionalAccountUpdates.get(address);
-		account.setMemo(memo);
-		account.setKey(key);
-		account.setProxyAccount(proxyAccount.asEntityId());
-
-		this.provisionalAccountUpdates.put(address, account);
-	}
-
 	public class WorldStateAccount implements Account {
```

Comment:
nit: The following constructor is not used anymore: `WorldStateAccount(final Address address, final Wei balance)` as well as: `getProxyAccount` and `setProxyAccount`